core/templates/domain/story/story-summary.model.ts

```diff
@@ -95,11 +95,17 @@ export class StorySummary {
   }
 
   getTopicUrlFragment(): string {
-    return this._topicUrlFragment;
+    if (this._topicUrlFragment !== undefined) {
+      return this._topicUrlFragment;
+    }
+    throw new Error('Topic Url Fragment is undefined');
```

Comment:
Do we throw errors in get methods in models normally? We expect validation methods to do these checks and report errors in the appropriate place.

Reply:
@kevintab95 these properties are optional and the methods were added with PR https://github.com/oppia/oppia/pull/13006 in the codebase, I don't find any particular usage of these methods other than tests, so I think it would be better if we will throw error here only.

Reply:
On a second thought, we can actually throw errors at component files where the methods will be used( Most probably with the project Redesigning and Updating Learner's Dashboard Page these functions will be used) at some component. I'm not very sure about this ,wdyt will be the right approach ?

Reply:
> Do we throw errors in get methods in models normally? We expect validation methods to do these checks and report errors in the appropriate place.

I think it's fine to keep validation here. If this skips some sort of validation and ends up here, we would want to know. But, this should not be here in a getter. You should do this check in the setter. (If undefined for _topicUrlFragment is not a valid value, it shouldn't be allowed to set to undefined in the first place. Can you throw an error there?)

Reply:
@srijanreddy98 These methods haven't been used anywhere yet, so no setters and at rest of the places this property will be undefined because of no usage like before

Reply:
This is now resolved we're actually returning a '#' at `getStoryLink` with usage of these two methods i.e. `getTopicUrlFragment` and `getClassroomUrlFragment` (if they return an undefined value) at `learner-story-summary-tile.component.ts` with this PR https://github.com/oppia/oppia/pull/13241. So no throwing of errors at getters. @krishita30j It'd be great if you can confirm this ?